edge-modules/api-proxy-module/src/monitors/config_parser.rs

```diff
@@ -93,19 +94,127 @@ impl ConfigParser {
         let config: String =
             envsubst::substitute(config, &self.context).context("Failed to subst the text")?;
 
+        // Find all regular expressions in if_tag and resolve them
+        let re = Regex::new(r"boolean_expression\[(.*)\]")
+            .context("Failed to match boolean statement in if_tag")?;
+
+        let config = re.replace_all(&config, |caps: &Captures| {
+            parse_boolean_expression(&caps[1])
+        });
+
         //Replace is 0
         let re = Regex::new(r"#if_tag 0((.|\n)*?)#endif_tag 0")
             .context("Failed to remove text between #if_tag 0 tags ")?;
         let config = re.replace_all(&config, "").to_string();
 
-        //Or not 1. This allows usage of if ... else ....
-        let re = Regex::new(r"#if_tag ![^0]((.|\n)*?)#endif_tag [^0].*?\n")
-            .context("Failed to remove text between #if_tag 0 tags ")?;
-        let config = re.replace_all(&config, "").to_string();
-
         Ok(config)
     }
 }
+enum Expr {
+    Value(bool),
+    Operator(char),
+}
+
+fn parse_boolean_expression(expression: &str) -> String {
+    let expression = replace_env_var_with_boolean(expression);
+    match solve_boolean_expression(&expression) {
+          Ok(x) => x,
+          Err(e) => { 
+              error!("{}", e); 
+              expression 
+            }
+     }    
+}
+
+// Gomma in env var name is forbidden because of that
+// We use this instead of simpler regex because we cannot match overlapping regular expression with regex.
+fn replace_env_var_with_boolean(expression: &str) -> String {
+    let mut fifo: Vec<char> = Vec::new();
+    let mut flush_fifo: Vec<char> = Vec::new();
+
+    for c in expression.chars() {
+        match c {
+            '(' | '!' | '&' | '|' => fifo.push(c),
+            ')' | ',' => {
+                if flush_fifo.len() == 1 && Some('0') == flush_fifo.pop() {
+                    fifo.push('0')
+                } else if flush_fifo.len() > 1 {
+                    fifo.push('1')
+                };
+                fifo.push(c);
+                flush_fifo.clear();
+            }
+            _ => flush_fifo.push(c),
+        }
+    }
+
+    fifo.iter().collect()
```

Comment:
cool! I didn't think we can collect into a string!

Reply:
Yes convenient :)